.github/workflows/automerge-dependabot.yaml

```diff
@@ -15,13 +15,21 @@ jobs:
         uses: dependabot/fetch-metadata@v1
         with:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
+      - name: Checkout
+        if: contains(steps.metadata.outputs.dependency-names, 'k8s.io/api')
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
       - name: Generate Documentation/compatibility.md
         if: contains(steps.metadata.outputs.dependency-names, 'k8s.io/api')
         run: |
           make generate --always-make
-          git add Documentation/compatibility.md
-          git commit -m "Generate Documentation/compatibility.md"
-          git push
+          changedFiles=$(git diff --name-only | wc -l)
```

Comment:
you can do `git diff --exit-code`

Reply:
but that would make the workflow fail, wouldn't it?

What I was thinking here was that we don't want to push anything if there aren't any changes to commit (An empty commit has exit code != 0)